src/lambda_function.py

```diff
@@ -15,136 +15,186 @@
+                                                'check URL protocol')
     if (not url.startswith('https://') and not url.startswith('http://')):
         url = 'https://' + url
     xray_recorder.end_subsegment()
 
     logger.info(f'LambdaFunction: Validating URL...')
-    subsegment = xray_recorder.begin_subsegment('lambda_function: validate URL')
+    subsegment = xray_recorder.begin_subsegment('lambda_function: '
+                                                'validate URL')
     if (not validators.url(url)):
         logger.info(f'LambdaFunction: URL is not valid.')
-        return {"error" : "The url was bad"}
+        return {"error": "The url was bad"}
     xray_recorder.end_subsegment()
 
-    #### Define the object skeleton
+    """
+    Define the object skeleton
+    """
     object = {
-        "url" : url,
-        "results" : []
+        "url": url,
+        "results": []
     }
 
-    # CredibilityScore
+    """
+    CredibilityScore
+    """
     logger.info(f'LambdaFunction: Trying to get credibility score...')
-    credibilityresult = {}
+    credibility_result = {}
     try:
-        credibilityresult = cr.getCredibilityScore(url)
-        object['results'].append(credibilityresult)
+        credibility_result = cr.get_credibility_score(url)
+        object['results'].append(credibility_result)
     except Exception as e:
         logger.info(f'LambdaFunction: Could not get Credibility Score.')
         logger.info(e)
-        credibilityresult = {'type': 'credibility', 'outcome': {"error" : "The credibility score was not available."}}
-        object['results'].append(credibilityresult)
-
-
-    # SentimentAnalisys
+        credibility_result = {'type': 'credibility',
+                              'outcome': {"error": 'The credibility score '
+                                                   'was not available.'}}
+        object['results'].append(credibility_result)
+
+    """
+    SentimentAnalisys
+    """
     logger.info(f'LambdaFunction: Trying to get sentimentAnalysis score...')
-    sentanalysisresult = {}
+    sentresult = {}
     try:
-        sentanalysisresult = sa.sentimentAnalysis(url)
-        if sentanalysisresult['text'] == '-1':
-            logger.info(f'LambdaFunction: sentimentAnalysis returned -1, dumping:')
-            logger.info(sentanalysisresult)
-            object['article'] = {'error': "The article summary could not be generated"}
-            object['results'].append({'type': 'polarity',     "outcome": {"error" : "The polarity score could not be calculated."}})
-            object['results'].append({'type': 'objectivity', "outcome": {"error" : "The objectivity score could not be calculated."}})
+        sentresult = sa.sentiment_analysis(url)
+        if sentresult['text'] == '-1':
+            logger.info(f'LambdaFunction: sentimentAnalysis returned -1, '
+                        'dumping:')
+            logger.info(sentresult)
+            object['article'] = {'error':
```

Comment:
It looks like you end up creating a similar object dictionary in a few different places and it looks pretty bulky having it inside this if statement. Perhaps you could create the error dict somewhere else in this file and then just do something like: `object['results] = failed_polarity_result` here?